piff/knn_interp.py

```diff
@@ -0,0 +1,373 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: knn_interp
+"""
+
+from __future__ import print_function
+
+from .interp import Interp
+from .starfit import Star, StarFit
+from .stardata import StarData
+
+import numpy
+import fitsio
+
+
+class kNNInterp(Interp):
+    """
+    An interpolator that uses sklearn KNeighborsRegressor to interpolate a
+    single surface
+    """
+    def __init__(self, logger=None, **knr_kwargs):
+        """Create the kNN interpolator
+
+        :param attr_interp: A list of star attributes to interpolate from
+        :param attr_target: A list of star attributes to interpolate to
+        :param logger:      A logger object for logging debug info. [default: None]
+        :param knr_kwargs:  Arguments that will be passed to the regressor.
+        """
+
+        self.kwargs = {
+            'n_neighbors': 15,
+            'weights': 'uniform',
+            'algorithm': 'auto',
+            'p': 2,
+            }
```

Comment:
These are not the construction kwargs.  Although probably these parameters ought to be added to the list of parameters to `__init__`, since they look like things we'd want to have as configurable options.